linode/instanceconfig/resource.go

```diff
@@ -270,26 +285,14 @@ func updateResource(ctx context.Context, d *schema.ResourceData, meta any) diag.
 			return diag.Errorf("failed to get config %d: %s", id, err)
 		}
 
-		bootedConfigID, err := helper.GetCurrentBootedConfig(ctx, &client, linodeID)
-		if err != nil {
-			tflog.Warn(
-				ctx, fmt.Sprintf("failed to get current booted config of Linode %d", linodeID),
-			)
-		}
-
-		powerOffRequired = instancehelpers.VPCInterfaceIncluded(config.Interfaces, putRequest.Interfaces) && id == bootedConfigID
+		powerOffRequired = instancehelpers.VPCInterfaceIncluded(config.Interfaces, putRequest.Interfaces) && isBootedConfig
 		shouldUpdate = true
 	}
 
 	// We should not use `HasChange(...)` here because of possible mid-apply changes
 	managedBoot := !d.GetRawConfig().GetAttr("booted").IsNull()
 
-	inst, err := client.GetInstance(ctx, linodeID)
-	if err != nil {
-		return diag.Errorf("Error finding the specified Linode Instance: %s", err)
-	}
-
-	shouldPowerBackOn := !managedBoot && powerOffRequired && inst.Status == linodego.InstanceRunning
```

Comment:
I think we should only power back on the instances with current status of `running` before we powering it off for VPC update?

Reply:
Oh, never mind, it was already included when `powerOffRequired` is created.